Add current page number in <title> and <h1>

welovewebdesign opened the pull request:
I added the current page number along with the word "page" in title and h1.
Taking effect automatically on every paged page.
What do you think ?

welovewebdesign committed:
Add current's page count in <title> and <h1>

bpierre commented on header.php:
```diff
@@ -9,12 +9,13 @@
 	<head>
 		<meta charset="<?php bloginfo('charset'); ?>" />
-		<title><?php wp_title('·', true, 'right'); ?> <?php bloginfo('name'); ?></title>
+		<title><?php wp_title('·', true, 'right'); if($paged) { echo " page $paged · "; } bloginfo('name'); ?></title>
```
( Thank you Rémi, really happy to see someone improving the theme :-) )

Maybe we could use this syntax: “Articles (page 2) · Les intégristes”, what do you think?

welovewebdesign replied:
Glad to contribute :)
Your syntax definitely looks better, i'll make the change!

bpierre commented on header.php:
```diff
@@ -33,7 +34,7 @@
 			</p>
 			<div>
-				<h1><a href="<?php echo get_option('home'); ?>/"><?php bloginfo('name'); ?></a></h1>
+				<h1><a href="<?php echo get_option('home'); ?>/"><?php bloginfo('name'); if($paged) { echo " · page $paged"; } ?></a></h1>
```
I think this one should be moved in the page header (`#content > h1`). The first `h1` is in a `<header role="banner">`, so I assumed it’s ok to keep the website title only here, whatever the current page is.

The problem is that the second one is located in several templates (maybe we could use a wrapper function everytime we need a page title, something like `lesintegristes_page_title("My original title")`).

Tell me if you agree, or not :-)

kaneel replied:
clapclap LE remy :)

2012/10/5 Pierre Bertet notifications@github.com

> In header.php:
> 
> > @@ -33,7 +34,7 @@
> >                 <a href="#searchform">Accéder à la recherche</a>
> >             </p>
> >             <div>
> > -               <h1><a href="<?php echo get_option('home'); ?>/"><?php bloginfo('name'); ?></a></h1>
> > -               <h1><a href="<?php echo get_option('home'); ?>/"><?php bloginfo('name'); if($paged) { echo " · page $paged"; } ?></a></h1>
> 
> I think this one should be moved in the  page header (#content > h1). The
> first h1 is in a <header role="banner">, so I assumed it’s ok to keep the
> website title only here, whatever the current page is.
> 
> The problem is that the second one is located in several templates (maybe
> we could use a wrapper function everytime we need a page title, something
> like lesintegristes_page_title("My original title")).
> 
> Tell me if you agree, or not :-)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lesintegristes/lesintegristes-theme/pull/39/files#r1777890.

## 

// Guillaume *kaneel *Richard http://mynameiskaneel.com
// With many lower than 3

welovewebdesign replied:
@bpierre : I totally aggree with you on the `h1` thing.

What kind of operation would you like the `lesintegristes_page_title` function to perform?
I'm thinking something like this (could be extended as needed):

``` php
function lesintegristes_page_title($page_title, $paged) {
  $page_title = $page_title;
  if($paged) $page_title .= " (page $paged)";
  return $page_title;
}
```

My concern is that you would have to call it this way : `lesintegristes_page_title("My original title", $paged)`.
But I don't think this is a big issue, what are your thought?

@kaneel : 

![](http://i.imgur.com/0sQ9h.gif)

bpierre replied:
`is_paged()` [is really cheap](http://core.trac.wordpress.org/browser/tags/3.4.2/wp-includes/query.php#L3359), as is `lesintegristes_get_current_page_number()`. Maybe we can just:

``` php
function lesintegristes_page_title($page_title) {
  if(is_paged()) {
    $paged = lesintegristes_get_current_page_number();
    return "$page_title (page $paged)";
  }
  return $page_title;
}
```

welovewebdesign committed:
Better title handling

- automatically filter page's title (adding page's number if needed)
- build custom title via lesintegristes_page_title function

bpierre commented:
I let you merge this branch into `master` when you want to.

Thanks!
